src/main/kotlin/com/manimdsl/frontend/SemanticAnalysis.kt

```diff
@@ -94,8 +106,23 @@ class SemanticAnalysis {
         ctx: ParserRuleContext
     ) {
         val method = dataStructureType.getMethodByName(methodName)
```

Comment:
It is because of the context types they don't match up 😅 but I can overload it

Reply:
Also this is an assignment so the method variable is returned so I can't do much

Reply:
Are they not both `ParserRuleContexts`?

Reply:
> Also this is an assignment so the method variable is returned so I can't do much

I just meant could we not call the `getMethodByName` function inside `ManimParserVisitor`?

Reply:
As in this if statement is used to assign to a variable so the method needs to be returned

Reply:
Oh yes this would not work for the constructor hence the overloading. But the function is never used with just methodName so I can delete the overloaded one

Reply:
Okay if you can refactor this to be a single function I think we'll be good to go 👍